server/src/model/scd.ts

```diff
@@ -0,0 +1,191 @@
+import * as D from 'io-ts/Decoder';
+import { fold } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/function';
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import {
+  ArraySchema,
+  EnumSchema,
+  Schema
+} from '../common/schema';
+
+interface Scd {
+    scenarioConfigDefinition: ScdDefinition; 
+}
+  interface ScdDefinition {
+    name: string;
+    documents: Array<ScdConfiguration>;
+  }
+
+  interface ScdConfiguration {
+    configurations: Array<ConfigurationDefinition>;
+  }
+
+  interface ConfigurationDefinition{
+    name: string; 
+    version: string; 
+    schemaversion: string; 
+    context: string; 
+    settings: Array<ScdSetting>; 
+  }
+
+  interface ScdSetting {
+    name: string; 
+    defaultvalue: string, 
+    datatype: string; 
+ }
+ 
+const ScdSetting: D.Decoder<unknown, ScdSetting> = D.struct({
+  name: D.string, 
+  defaultvalue: D.string, 
+  datatype: D.string, 
+});
+
+const ConfigurationComponent: D.Decoder<unknown, ConfigurationDefinition> = D.struct({
+  name: D.string, 
+  version: D.string, 
+  schemaversion: D.string, 
+  context: D.string, 
+  settings: D.array(ScdSetting),
+});
+  
+const ScdConfiguration: D.Decoder<unknown, ScdConfiguration> = D.struct({
+  configurations: D.array(ConfigurationComponent),
+});
+
+const ScdComponents: D.Decoder<unknown, ScdDefinition> = D.struct({
+  name: D.string, 
+  documents: D.array(ScdConfiguration),
+});
+  
+const Scd: D.Decoder<unknown, Scd> = D.struct({
+  scenarioConfigDefinition: ScdComponents, 
+});
+
+function stringLiteral(value: string): EnumSchema
+{
+  return {
+    type: 'enum', 
+    valueSchema: 'string', 
+    enumValues: [
+      {
+        name: '', 
+        enumValue: value, 
+      }
+    ]
+  };
+}
+
+function settingsToSchema(scd: Scd): Schema {
+  return {
+    type: 'object', 
+    fields: scd.scenarioConfigDefinition.documents[0].configurations[0].settings.map((component) => {
+      return {
+        name: component.name,
+        schema: stringLiteral(component.defaultvalue),
+      };
+    })
```

Comment:
nit: will something like this work?
```suggestion
    fields: scd.scenarioConfigDefinition.documents[0].configurations[0].settings.map((component) => {
      name: component.name,
      schema: stringLiteral(component.defaultvalue),
    }
   })
```

Reply:
If it does, it may be worth changing in `mim.ts` too

Reply:
I can make those changes in mim.ts as well